pkg/pool-manager/mac-entry.go

```diff
@@ -0,0 +1,38 @@
+package pool_manager
+
+import "time"
+
+func (m macEntry) isPending() bool {
+	if m.transactionTimestamp != nil {
+		return true
+	}
+	return false
+}
+
+func (m macEntry) addTime(duration time.Duration) time.Time {
+	return m.transactionTimestamp.Add(duration)
+}
```

Comment:
Added a comment later on so we include this at different function.

Reply:
did catch your drift